### PR TITLE
feat: Open links in docs panel if enabled

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Part of the flagged feature of docs in the side panel - opens all `posthog.com` links there. 

## Changes

* Adds this handler to links
* Removes the "draggable" property from links as they are anyways draggable and it unlocks selection again
* Fixed a typo in the referrer check for posthog.com links

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by daibhin — APPROVED** [review]

File: frontend/src/layout/navigation-3000/sidepanel/panels/sidePanelDocsLogic.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { actions, kea, reducers, path, listeners, connect } from 'kea'
+import { SidePanelTab, sidePanelLogic } from '../sidePanelLogic'
+
+import type { sidePanelDocsLogicType } from './sidePanelDocsLogicType'
+
+const posthogComDomain = 'https://posthog.com'
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/panels/sidePanelDocsLogic.ts:6]: Nit: constants should be uppercase, e.g. `POSTHOG_COM_DOMAIN`… or `WEBSITE_DOMAIN`

File: frontend/src/scenes/notebooks/AddToNotebook/DraggableToNotebook.tsx [hunk 2]
```diff
@@ -81,6 +80,7 @@ export function DraggableToNotebook({
             <FlaggedFeature flag={FEATURE_FLAGS.NOTEBOOKS} fallback={children}>
                 <span
                     className={clsx('DraggableToNotebook', className, isDragging && 'DraggableToNotebook--dragging')}
+                    draggable={elementProps.onDragEnd ? true : false}
```
Comment (Twixes) [inline frontend/src/scenes/notebooks/AddToNotebook/DraggableToNotebook.tsx:83]: Is this relevant to the docs panel?

File: frontend/src/lib/lemon-ui/Link/Link.tsx [hunk 3]
```diff
@@ -43,7 +46,11 @@ const shouldForcePageLoad = (input: any): boolean => {
 }
 
 const isPostHogDomain = (url: string): boolean => {
```
Comment (Twixes) [inline frontend/src/lib/lemon-ui/Link/Link.tsx:48]: I guess this should be `isPostHogCloudDomain` if there is also `isPostHogComDomain`

**Review by Twixes — COMMENTED** [review]
This is really cool, I have a few nits and CodeQL makes a good point about URL checking, but overall definitely good to go!

File: frontend/src/scenes/notebooks/AddToNotebook/DraggableToNotebook.tsx (hunk 2 above)
Comment (benjackwhite) [reply frontend/src/scenes/notebooks/AddToNotebook/DraggableToNotebook.tsx:83]: This is only semi related. I was trying to select a link and I couldn't, because we override the draggable. I can pull this out to a different PR

File: frontend/src/lib/lemon-ui/Link/Link.tsx (hunk 3 above)
Comment (benjackwhite) [reply frontend/src/lib/lemon-ui/Link/Link.tsx:48]: Not quite, the first one is used to determine if it generally is a domain of ours, the other is explicitly the marketting website
